eth/protocols/snap/sync.go

```diff
@@ -622,6 +629,7 @@ func (s *Syncer) loadSyncStatus() {
 				log.Debug("Scheduled account sync task", "from", task.Next, "last", task.Last)
 			}
 			s.tasks = progress.Tasks
+			s.snapped = len(s.tasks) == 0
```

Comment:
This might be a dumb comment, but everywhere else we need to hold a lock for accessing `s` especially `s.snapped`. The only instance I can find where `loadSyncStatus()` is called is in Line 528, but we don't hold the lock there anymore

Reply:
Yeah it's a bit funky. `loadSyncStatus` touches a lot of internals in an unsafe way, and right after, the code accesses `s.tasks`. 

```
	s.loadSyncStatus()
	if len(s.tasks) == 0 && s.healer.scheduler.Pending() == 0 {
``` 
So probably that lock-holding should be extended to cover that portion too. 

I don't think it's a real issue, however, because I don't think we enter there in a concurrent way.

Reply:
AFAIK loadSyncStatus is only called when you create the thing ,so there's no concurrency at that point.